### PR TITLE
Fix exit codes on failures.

### PR DESCRIPTION
**Pull request by Tuxdude**
Chain through correct exit codes on failures in build.py and TestRunner.java.

### CODE REVIEW

**Comment by Tuxdude** [comment]
Here is one example where the builds were failing but build.py was ignoring it. https://travis-ci.org/EVelez79/CodeU-ProjectGroup6/jobs/241061629

File: test/codeu/chat/TestRunner.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public static void main(String[] args) {
          );
       for (final Failure failure : result.getFailures()) {
          System.out.println(failure.toString());
+         System.exit(-1);
```
Comment (vaage) [inline test/codeu/chat/TestRunner.java]: The test runner should print all the failures before exiting. Can you move this below `System.out.println(result.wasSuccessful());` and use `result.wasSuccessful()` to determine what the return value exit code should be?

File: build.py [hunk 2]
```diff
@@ -98,7 +98,12 @@ def build(config) :
   command += src_files
 
   print('running : %s' % command)
-  print('Build %s' % ('PASSED' if subprocess.call(command) == 0 else 'FAILED'))
+  if subprocess.call(command) != 0:
```
Comment (vaage) [inline build.py]: This is a NIT but could you put a blank line between 100 and 101 - just like you have in under RUN?

**Comment by Tuxdude** [comment]
Amended the commit.

File: test/codeu/chat/TestRunner.java [hunk 3]
```diff
@@ -33,6 +33,11 @@ public static void main(String[] args) {
       for (final Failure failure : result.getFailures()) {
          System.out.println(failure.toString());
       }
+
+      if (!result.wasSuccessful()) {
```
Comment (Tuxdude) [inline test/codeu/chat/TestRunner.java]: Done. `System.exit(0);` is redundant, so did not add that.

File: test/codeu/chat/TestRunner.java [hunk 4]
```diff
@@ -30,14 +30,14 @@ public static void main(String[] args) {
              codeu.chat.util.UuidTest.class,
              codeu.chat.util.store.StoreTest.class
          );
-      for (final Failure failure : result.getFailures()) {
-         System.out.println(failure.toString());
-      }
 
-      if (!result.wasSuccessful()) {
+      if (result.wasSuccessful()) {
+        System.out.println(result.wasSuccessful());
```
Comment (vaage) [inline test/codeu/chat/TestRunner.java]: As just printing "TRUE" is not that meaningful to a user (it has bothered me for a long time) can we change it to be a more meaningful message?

File: test/codeu/chat/TestRunner.java [hunk 5]
```diff
@@ -30,14 +30,14 @@ public static void main(String[] args) {
...
+      } else {
+        for (final Failure failure : result.getFailures()) {
+           System.out.println(failure.toString());
+        }
```
Comment (vaage) [inline test/codeu/chat/TestRunner.java]: Can we add a header to give context to the failure messages?

File: test/codeu/chat/TestRunner.java (hunk 4 above)
Comment (Tuxdude) [inline test/codeu/chat/TestRunner.java]: Done.

File: test/codeu/chat/TestRunner.java (hunk 5 above)
Comment (Tuxdude) [inline test/codeu/chat/TestRunner.java]: Done.

**Comment by wgenemorgan** [comment]
Waiting for Aaron's final approval.

**Pull request merged**